### PR TITLE
Add request fetching pickup points list

### PR DESCRIPTION
**Pull request by SzymonKaminski**

### CODE REVIEW

File: src/DTO/PickupPoint.php [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+    /**
+     * @var string
+     */
+    public $coordinateY;
+
+    /**
+     * @var string
+     */
+    public $coordinateX;
```
Comment (malarzm) [inline src/DTO/PickupPoint.php]: Coodrinates could use a `Coordinate` VO with `x` and `y` fields

Comment (malarzm) [reply src/DTO/PickupPoint.php]: Also why string and not float?

File: src/Factory/PickupPointFactory.php [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+            (string) $pickupPoint[self::POVIAT],
+            (string) $pickupPoint[self::COMMUNE],
+            (string) $pickupPoint[self::NAME],
+            (string) $pickupPoint[self::POSTCODE],
+            (string) $pickupPoint[self::LOCALITY],
+            (string) $pickupPoint[self::STREET],
+            (string) $pickupPoint[self::PHONE_NUMBER],
+            (string) $pickupPoint[self::DESCRIPTION],
+            (string) $pickupPoint[self::STATUS],
+            (string) $pickupPoint[self::TYPE]
```
Comment (malarzm) [inline src/Factory/PickupPointFactory.php:81]: Isn't this all `string`s by default?

File: src/Request/FetchPickupPointsRequest.php [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+class FetchPickupPointsRequest
+{
+    private const ZIP_URL = 'https://odbiorwpunkcie.poczta-polska.pl/pliki.php?t=xmlK48S';
+    private const ZIP_LOCAL_FILENAME = 'poczta-polska-pickup-points.zip';
+
+    /**
+     * @var ConfigProvider
+     */
+    private $configProvider;
+    /**
```
Comment (malarzm) [inline src/Request/FetchPickupPointsRequest.php:22]: missing newline

File: src/Factory/PickupPointFactory.php (hunk 2 above)
Comment (SzymonKaminski) [reply src/Factory/PickupPointFactory.php:81]: I followed https://stackoverflow.com/questions/2867575/get-value-from-simplexmlelement-object
Coudln't get it to work without explicit casting and combination of json_decode and json_encode mentioned there didn't work as intended, still leaving objects inside array.

**Review by malarzm — APPROVED** [review]

File: src/Request/FetchPickupPointsRequest.php [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+namespace Answear\PocztaPolskaBundle\Request;
+
+use Answear\PocztaPolskaBundle\DTO\PickupPoint;
+use Answear\PocztaPolskaBundle\Factory\PickupPointFactory;
+use Answear\PocztaPolskaBundle\Service\ConfigProvider;
+use ZipArchive;
+
+class FetchPickupPointsRequest
+{
+    private const ZIP_URL = 'https://odbiorwpunkcie.poczta-polska.pl/pliki.php?t=xmlK48S';
```
Comment (bmulawa) [inline src/Request/FetchPickupPointsRequest.php:14]: This will be configurable, right?

File: tests/Functional/Factory/PickupPointFactoryTest.php [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+    /**
+     * @test
+     */
+    public function createsArrayFrom(): void
+    {
+        $factory = $this->getFactory();
+
+        $pickupPoints = $factory->createArrayFromXml(__DIR__ . '/../../DataFixtures/XML/pickupPoints.xml');
+
+        $this->assertCount(3, $pickupPoints);
```
Comment (bmulawa) [inline tests/Functional/Factory/PickupPointFactoryTest.php]: This assertion is not needed as next one will fail anyway - with more details.

File: src/Request/FetchPickupPointsRequest.php (hunk 4 above)
Comment (SzymonKaminski) [reply src/Request/FetchPickupPointsRequest.php:14]: Its not needed, url doesnt change

Comment (bmulawa) [reply src/Request/FetchPickupPointsRequest.php:14]: Are you sure? `t` parameter seems like some id of current content

**Review by bmulawa — APPROVED** [review]

**Pull request merged**